components/backends/trtllm/src/dynamo/trtllm/publisher.py

```diff
@@ -394,6 +409,24 @@ async def cleanup(self):
             if self.publish_kv_cache_events_thread.is_alive():
                 logging.warning("KV cache events thread did not stop within timeout")
 
+    def update_max_window_size(self, event):
+        if "window_size" in event:
+            window_size = event["window_size"]
+            if self.max_window_size is None or window_size > self.max_window_size:
+                self.max_window_size = window_size
+                logging.debug(
+                    f"kv events max_window_size has been updated to {self.max_window_size}"
+                )
+
+    def should_drop_event(self, event):
+        if "window_size" not in event or self.processing_initial_created_events:
```

Comment:
Can you add a line here about how global attention layer is related to `window_size` and `self.processing_initial_created_events`?

Reply:
It is bit hard to parse.